client/wp-admin-scripts/print-shipping-label-banner/shipping-banner/index.js

```diff
@@ -0,0 +1,448 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import { Button, ExternalLink } from '@wordpress/components';
+import { compose } from '@wordpress/compose';
+import { recordEvent } from 'lib/tracks';
+import interpolateComponents from 'interpolate-components';
+
+/**
+ * Internal dependencies
+ */
+import '../style.scss';
+import DismissModal from '../dismiss-modal';
+import { getSetting } from '@woocommerce/wc-admin-settings';
+import withSelect from 'wc-api/with-select';
+import SetupNotice, { setupErrorTypes } from '../setup-notice';
+import { withDispatch } from '@wordpress/data';
+import { getWcsAssets, acceptWcsTos } from '../wcs-api';
+
+const wcAdminAssetUrl = getSetting( 'wcAdminAssetUrl', '' );
+
+export class ShippingBanner extends Component {
+	constructor( props ) {
+		super( props );
+
+		const orderId = new URL( window.location.href ).searchParams.get(
+			'post'
+		);
+
+		this.state = {
+			showShippingBanner: true, // TODO: update to get state when closedForever is clicked
+			isDismissModalOpen: false,
+			setupErrorReason: setupErrorTypes.SETUP,
+			orderId: parseInt( orderId, 10 ),
+			wcsAssetsLoaded: false,
+			wcsAssetsLoading: false,
+			wcsSetupError: false,
+			isShippingLabelButtonBusy: false,
+		};
+	}
+
+	componentDidMount() {
+		this.trackImpression();
+	}
+
+	componentDidUpdate( prevProps ) {
+		const { activatePlugins, wcsPluginSlug } = this.props;
+
+		if ( this.justInstalledWcs( prevProps ) ) {
+			activatePlugins( [ wcsPluginSlug ] );
+		}
+		if ( this.justActivatedWcs( prevProps ) ) {
+			// TODO: Add success notice after installation #32
+			this.acceptTosAndGetWCSAssets();
+		}
+	}
+
+	justInstalledWcs( prevProps ) {
+		const { installedPlugins, wcsPluginSlug } = this.props;
+		const wcsNowInstalled = installedPlugins.includes( wcsPluginSlug );
+		const wcsPrevInstalled = prevProps.installedPlugins.includes(
+			wcsPluginSlug
+		);
+		return wcsNowInstalled && ! wcsPrevInstalled;
+	}
+
+	justActivatedWcs( prevProps ) {
+		const { activatedPlugins, wcsPluginSlug } = this.props;
+		const wcsNowActivated = activatedPlugins.includes( wcsPluginSlug );
+		const wcsPrevActivated = prevProps.activatedPlugins.includes(
+			wcsPluginSlug
+		);
+		return wcsNowActivated && ! wcsPrevActivated;
+	}
+
+	hasActivationError = () => {
+		return Boolean( this.props.activationErrors.length );
```

Comment:
Do we need to verify this prop is present ( and an array ) - or perhaps we add some required propType declarations to help protect.

Reply:
Proptypes will take care of it